### PR TITLE
[#12048] E2E test migration

### PR DESCRIPTION
**Pull request by cedricongjh**
Part of #12048 
**Outline of Solution**
Create E2E test for SQL

### CODE REVIEW

File: src/test/java/teammates/architecture/ArchitectureTest.java [hunk 1]
```diff
@@ -343,7 +343,13 @@ public void testArchitecture_e2e_e2eShouldBeSelfContained() {
     @Test
     public void testArchitecture_e2e_e2eShouldNotTouchProductionCodeExceptCommon() {
         noClasses().that().resideInAPackage(includeSubpackages(E2E_PACKAGE))
-                .should().accessClassesThat().resideInAPackage(includeSubpackages(STORAGE_PACKAGE))
+                .should().accessClassesThat(new DescribedPredicate<>("") {
+                    @Override
+                    public boolean apply(JavaClass input) {
+                        return input.getPackageName().startsWith(STORAGE_PACKAGE)
+                                && !input.getPackageName().startsWith(STORAGE_SQL_ENTITY_PACKAGE);
```
Comment (ziqing26) [inline src/test/java/teammates/architecture/ArchitectureTest.java:350]: Just to double confirm: currently the architecture test need to access the whole package for datastore storage but not sql storage, but in the future we need to include sql storage package for architecture test?

**Review by ziqing26 — APPROVED** [review]
The base test and student home page test LGTM! Thanks for the work:)

**Review by weiquu — APPROVED** [review]
LGTM

File: src/test/java/teammates/architecture/ArchitectureTest.java (hunk 1 above)
Comment (cedricongjh) [reply src/test/java/teammates/architecture/ArchitectureTest.java:350]: ah it's actually the opposite, this test ensures that the E2E tests do not access the storage package, however the modification here allows the E2E tests to access only the sql entities in the storage package, as previously the entities (datastore) were not in the storage package

**Pull request merged**